src/tpg_tcp_sm.c

```diff
@@ -451,16 +451,34 @@ int tsm_dispatch_net_event(tcp_control_block_t *tcb, tcpEvent_t event,
 
     if (pcb->pcb_tcp != NULL) {
         TCB_TRACE(tcb, TSM, DEBUG, "   tcp {flags=%c%c%c%c%c%c, seq=%u, ack=%u, win=%u, urp=%u}",
-                  (pcb->pcb_tcp->tcp_flags & TCP_URG_FLAG) == 0 ? '-' : 'u',
-                  (pcb->pcb_tcp->tcp_flags & TCP_ACK_FLAG) == 0 ? '-' : 'a',
-                  (pcb->pcb_tcp->tcp_flags & TCP_PSH_FLAG) == 0 ? '-' : 'p',
-                  (pcb->pcb_tcp->tcp_flags & TCP_RST_FLAG) == 0 ? '-' : 'r',
-                  (pcb->pcb_tcp->tcp_flags & TCP_SYN_FLAG) == 0 ? '-' : 's',
-                  (pcb->pcb_tcp->tcp_flags & TCP_FIN_FLAG) == 0 ? '-' : 'f',
+                  (pcb->pcb_tcp->tcp_flags & RTE_TCP_URG_FLAG) == 0 ? '-' : 'u',
+                  (pcb->pcb_tcp->tcp_flags & RTE_TCP_ACK_FLAG) == 0 ? '-' : 'a',
+                  (pcb->pcb_tcp->tcp_flags & RTE_TCP_PSH_FLAG) == 0 ? '-' : 'p',
+                  (pcb->pcb_tcp->tcp_flags & RTE_TCP_RST_FLAG) == 0 ? '-' : 'r',
+                  (pcb->pcb_tcp->tcp_flags & RTE_TCP_SYN_FLAG) == 0 ? '-' : 's',
+                  (pcb->pcb_tcp->tcp_flags & RTE_TCP_FIN_FLAG) == 0 ? '-' : 'f',
                   rte_be_to_cpu_32(pcb->pcb_tcp->sent_seq),
                   rte_be_to_cpu_32(pcb->pcb_tcp->recv_ack),
                   rte_be_to_cpu_16(pcb->pcb_tcp->rx_win),
                   rte_be_to_cpu_16(pcb->pcb_tcp->tcp_urp));
+
+        if (pcb->pcb_tcp->tcp_flags & RTE_TCP_SYN_FLAG) {
```

Comment:
Same for these counters. Let's increment the stats later on in the stack.

Reply:
Agreed. Done.